docs/dev/contributing/backend/Guide-to-Creating-an-Endpoint.md

```diff
@@ -1,34 +1,207 @@
-
-It was written for external facing APIs, rather than internal ones. But most concepts should apply equally to internal APIs.
-
-This page is a skeleton that we hope to populate with more details. 
-
-* Add a new entry into the yaml
-	* URL design and structure: [API Style Guide](https://github.com/transcom/mymove/wiki/API-Style-Guide)
-	* Path Parameters, Headers and Body 
-	* Response Body
-		* Required
-		* x-nullable
-		* Readonly
-		* Documentation
-	* Error Responses: [API Errors Guide](https://github.com/transcom/mymove/wiki/API-Errors)
-* Add a handler for the endpoint
-	* Add payload_to_model and model_to_payload functions
-	* Create handler type and Handle function
-		* Hook it up to api.go
-	* How to handle errors
-* Add service object
-	* When to make a service object: [Service Objects](https://github.com/transcom/mymove/wiki/service-objects)
-	* Add service object interface
-	* Add service object constructor
-	* Add service object code
-	* Add service object tests
-* Add event key and update event map
-	* Each API has a corresponding file in `/pkg/services/event/<apiName>_endpoint.go`
-	* Add new `const` to represent event key
-	* Add event key to endpoint map in the same file, using the event key name, api name, and operation ID.
-* Add tests for the handler
-	* Add mocks: [Generating Mocks with mockery](https://github.com/transcom/mymove/wiki/generate-mocks-with-mockery)
-	* Add test code
-        * Use testdatagen functions [[Understanding Testdatagen Functions]]
+# Creating an Endpoint
+###### These are the various steps that are involved in creating a new endpoint.
+
+Prior to creating adding an endpoint to the Handler folder, we must first add a new endpoint definition to swagger. We are using Swagger 2.0, which is [OpenAPI](https://swagger.io/specification/v2/), a specification we use to format our RESTful APIs and provide a template for us to communicate the information in our API.
+Always start with swagger. This step creates your endpoint definition and generates the files and helper functions you will need to create your endpoint. More specifically, swagger converts JSON user input into generated Go types.
+## Adding  a new entry into the yaml
+All new definitions will be added in `mymove/swagger-def`. Note that we have broken down our spec into different files and we share definitions between files. 
+Built compiled versions of our API spec will be generated and stored in the swagger folder, which we will not edit directly. Notice that there is a yaml file for each of our APIs. 
+An endpoint definition for the prime will go into the Prime yaml, but you may notice there are some definitions in `mymove/swagger-def/definitions`.
+This is because some definitions are shared across APIs and we've created a space to add those definitions in one place. 
+
+For the purposes for adding a new endpoint, make sure that your endpoint is defined in these top level sections:
```

Comment:
I'm not sure top level sections is the correct term. You might say `And endpoint definition must include the following items` or something along those lines.